src/main/java/seedu/duke/logic/parser/AddCommandParser.java

```diff
@@ -68,69 +66,78 @@ private static Command parseAddLessonCommand(String userResponse) throws ParseEx
             throw new ParseException(Messages.ERROR_INVALID_TIME_FORMAT);
         }
 
-        String meetingUrl = "-";
         switch (params.length) {
         case 4:
-            if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e")) {
-                throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
-            }
-            return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+            return parseAddLessonFourParams(userResponse, title, dayOfTheWeek, startTime, endTime);
         case 5:
-            if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e", "-l")) {
-                throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
-            }
-            meetingUrl = params[4].strip();
-            return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+            return parseAddLessonFiveParams(userResponse, params[4], title, dayOfTheWeek, startTime, endTime);
         default:
             throw new ParseException(Messages.ERROR_INVALID_COMMAND);
         }
     }
 
+    private static Command parseAddLessonFiveParams(String userResponse, String param, String title,
+                                                    String dayOfTheWeek, String startTime, String endTime)
+            throws ParseException {
+        String meetingUrl;
+        if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e", "-l")) {
+            throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
+        }
+        meetingUrl = param.strip();
+        return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, meetingUrl);
+    }
+
+    private static AddLessonCommand parseAddLessonFourParams(String userResponse, String title, String dayOfTheWeek,
+                                                             String startTime, String endTime) throws ParseException {
+        if (!hasCorrectFlagSequence(userResponse, "-d", "-s", "-e")) {
+            throw new ParseException(Messages.ERROR_INVALID_FLAG_SEQUENCE);
+        }
+        return new AddLessonCommand(title, dayOfTheWeek, startTime, endTime, EMPTY_INFORMATION);
+    }

```

Comment:
Hmm.. I don't think this is a particularly better implementation than the previous commits. Consider combining these two methods into one, and move them to `ParserUtil`.